### PR TITLE
Refactor MTE-3854 Integrate menu refactor work and xcode 16

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3854)

## :bulb: Description
Cherry pick menu refactor work after Xcode 16 is in use.

TODO
* Changes required for full functional tests

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan:122]: This test fails with the menu refactor turned on. 😞

**Comment by clarmso** [comment]
@isabelrios Should we merge this PR just with the changes for smoke tests? 🙂 Please approve this PR if it's the case. We can address the full functional tests in a subsequent PR.
(cc @thatswinnie)

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ToolbarMenuTests.swift
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ToolbarMenuTests.swift:50]: can we add a comment why we comment these lines?

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ToolbarMenuTests.swift
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ToolbarMenuTests.swift:65]: same as above

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/TrackingProtectionTests.swift
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/TrackingProtectionTests.swift:214]: Please do not change the indentation (unless it's swiftlint related).

Comment (adudenamedruby) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/TrackingProtectionTests.swift:214]: I don't understand this indentation. It's not very Swifty. lolol oh well

**Review by isabelrios — APPROVED** [review]
Please address comment about swiftlint here or in a follow up

**Comment by clarmso** [comment]
Branch with menu refactor off before merging this PR: https://github.com/mozilla-mobile/firefox-ios/tree/release/NoMenuRefactorDec2

**Review by mdotb-moz — APPROVED** [review]
LGTM

**Pull request merged**
